src/LINEBot/HTTPClient/CurlHTTPClient.php

```diff
@@ -158,6 +182,12 @@ private function getOptions($method, $headers, $reqBody)
                 $options[CURLOPT_POSTFIELDS] = $reqBody;
             }
         }
+        if (!is_null($this->timeout)) {
+            $options[CURLOPT_TIMEOUT] = $this->timeout;
```

Comment:
Late curl have CURLOPT_CONNECTTIMEOUT_MS and CURLOPT_TIMEOUT_MS. However, this repository covers php5 that cannot use curl with _MS options.

Reply:
We are testing curl requests by invoking actual http request to `req_mirror.php` server.
Please see `CurlHTTPClientTest`.

Reply:
Should I add delaying response feature to `req_mirror.php` to testing request timeout settings?

Reply:
Yes, how about taking `responseDelay` url parameter, so that clients can change server side response time.

Reply:
Thank you. I added `responseDelay` url parameter to `req_mirror.php` and add timeout curl tests.